pygmt/tests/test_timestamp.py

```diff
@@ -0,0 +1,88 @@
+"""
+Tests for Figure.timestamp.
+"""
+import pytest
+from packaging.version import Version
+from pygmt import Figure, __gmt_version__
+
+
+@pytest.fixture(scope="module", name="faketime")
+def fixture_faketime():
+    """
+    Fake datetime that will be passed to the "timefmt" parameter, so that the
+    timestamp string always has a fixed value.
+    """
+    return "9999-99-99T99:99:99"
+
+
+@pytest.mark.mpl_image_compare
+def test_timestamp(faketime):
+    """
+    Test that the simplest timestamp() call works.
+    """
+    fig = Figure()
+    fig.timestamp(timefmt=faketime)
+    return fig
+
+
+@pytest.mark.mpl_image_compare
+def test_timestamp_label(faketime):
+    """
+    Check if the "label" parameter works.
+    """
+    fig = Figure()
+    fig.timestamp(label="Powered by PyGMT", timefmt=faketime)
+    return fig
+
+
+@pytest.mark.mpl_image_compare
+def test_timestamp_justification():
+    """
+    Check if the "justification" parameter works.
+
+    Only a subset of justification codes are tested to avoid overlapping
+    timestamps.
+    """
+    fig = Figure()
+    fig.basemap(projection="X10c/5c", region=[0, 10, 0, 5], frame=0)
+    for just in ["BL", "BR", "TL", "TR"]:
+        fig.timestamp(justification=just, timefmt=just)
+    return fig
```

Comment:
The output of the justification image doesn't look correct?

![](https://camo.githubusercontent.com/7315bdbdb236530531f923d1ed7811d1ec4f71ce456bad71fefcae41267a47ff/68747470733a2f2f61737365742e636d6c2e6465762f303461623234646433643639363434626365633562643431666165633065393165333162346238343f636d6c3d706e672663616368652d6279706173733d38396366636162302d323930662d346434392d623439622d393234333964663835366435)

All of the timestamps are on the bottom left (BL). Shouldn't they be at the four different corners?

Reply:
Unlike other embellishments like inset, legend or direction rose, the timestamp logo are always located at the lower left corner of the map. I believe this is GMT's design.

See https://docs.generic-mapping-tools.org/dev/gmt.conf.html#term-MAP_LOGO_POS

> Sets the justification and the position of the logo/timestamp box **relative to the current plot’s lower left corner** (i.e., map origin)

Reply:
Interesting design choice :sweat_smile: Actually, I just realized I was thinking of `position` rather than `justification`. Maybe we should document the default `offset="-54p/-54p"` under the offset docstring?

Reply:
The default values are clear from the function definition:

`Figure.timestamp((text=None, label=None, justification='BL', offset=('-54p', '-54p'), font='Helvetica,black', timefmt='%Y %b %d %H:%M:%S')`

I'm hesitate to document the defaults in the docstring.

Reply:
Ah ok, was thinking of adding `[Default is ('-54p', '-54p')]`, but it's not really needed I guess.